From faec1df983dc6864d80ff01035882cfe3b33640e Mon Sep 17 00:00:00 2001
From: Erik Schumacher <erik.schumacher@iris-sensing.com>
Date: Fri, 5 Nov 2021 13:59:48 +0100
Subject: [PATCH] Fix compilation of fsl-image-mfgtool-initramfs

Set different BOOT_TOOLS directory for mfgtools recipe
Use SPL_BINARYNAME to generate the deployment file name
See also: https://github.com/Freescale/meta-freescale/commit/c72cb38b891cb916d3d71d86bfd0787ba947de2c
---
 recipes-bsp/u-boot/u-boot-mfgtool.inc | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/recipes-bsp/u-boot/u-boot-mfgtool.inc b/recipes-bsp/u-boot/u-boot-mfgtool.inc
index 91526ad7..96651b75 100644
--- a/recipes-bsp/u-boot/u-boot-mfgtool.inc
+++ b/recipes-bsp/u-boot/u-boot-mfgtool.inc
@@ -17,5 +17,8 @@ UBOOT_CONFIG = "mfgtool"
 # Add 'mfgtool' suffix
 UBOOT_IMAGE = "u-boot-${MACHINE}-mfgtool-${PV}-${PR}.${UBOOT_SUFFIX}"
 UBOOT_SYMLINK = "u-boot-${MACHINE}-mfgtool.${UBOOT_SUFFIX}"
-SPL_IMAGE = "${SPL_BINARY}-${MACHINE}-mfgtool-${PV}-${PR}"
-SPL_SYMLINK = "${SPL_BINARY}-mfgtool-${MACHINE}"
+SPL_IMAGE = "${SPL_BINARYNAME}-${MACHINE}-mfgtool-${PV}-${PR}"
+SPL_SYMLINK = "${SPL_BINARYNAME}-mfgtool-${MACHINE}"
+
+# Let 'mfgtool' deploy to a different directory to avoid conflicts
+BOOT_TOOLS = "imx-mfgtool"
-- 
2.31.1

